File: iosApp/iosApp/Bridge/SongBridgeImpl.swift
import ComposeApp

class SongBridgeImpl: SongBridge {
    private let songRepository = SongRepository()

    func findAll() -> [SongModel] {
        return songRepository.findAll()
    }

    func findById(id: String) -> SongModel? {
        return songRepository.findById(id: id)
    }
}

File: composeApp/src/androidMain/kotlin/dev/tcode/thinmpk/bridge/SongBridgeImpl.kt
package dev.tcode.thinmpk.bridge

import dev.tcode.thinmpk.model.SongModel
import dev.tcode.thinmpk.repository.SongRepository

class SongBridgeImpl : SongBridge {
    private val songRepository = SongRepository()

    override fun findAll(): List<SongModel> {
        return songRepository.findAll()
    }

    override fun findById(id: String): SongModel? {
        return songRepository.findById(id)
    }
}
